server/config.go

```diff
@@ -93,22 +94,22 @@ func NewConfiguration(env Environment) (*Config, error) {
 		return nil, err
 	}
 
-	if configSettings.Server.Port != 0 {
+	if env.IsSet("server.port") && configSettings.Server.Port != 0 {
```

Comment:
I would suggest droping the IsSet method and change 
lines 92-96 (the configSettings in NewConfiguration) as follows : 
```
configSettings := &Settings{
		Server: &AppSettings{},
		Db:     &DbSettings{},
		Log:    &LogSettings{},
	}
```

The whole purpose of doing unmarshal is not not hardcode stuff like "server.port" in the code but use struct fields only